### PR TITLE
TW-1380: Update Applifecycle for twake web

### PR DESCRIPTION
**Pull request by sherlockvn**

### CODE REVIEW

File: lib/widgets/matrix.dart [hunk 1]
```diff
@@ -591,9 +594,17 @@ class MatrixState extends State<Matrix>
     }
   }
 
+  void onFocus(html.Event e) {
+    didChangeAppLifecycleState(AppLifecycleState.resumed);
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:598]: please check only in web

File: lib/widgets/matrix.dart [hunk 2]
```diff
@@ -591,9 +594,17 @@ class MatrixState extends State<Matrix>
...
+  }
+
+  void onBlur(html.Event e) {
+    didChangeAppLifecycleState(AppLifecycleState.paused);
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:602]: please check only in web

File: lib/widgets/matrix.dart [hunk 3]
```diff
@@ -591,9 +594,17 @@ class MatrixState extends State<Matrix>
...
+  }
+
   @override
   void didChangeAppLifecycleState(AppLifecycleState state) {
-    Logs().v('AppLifecycleState = $state');
+    debugPrint('AppLifecycleState = $state');
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:607]: add timestamp to get more detail in log

**Review by hoangdat — COMMENTED** [review]
Also need to add timestamp in log of `/sync`

**Comment by sherlockvn** [comment]
updated:
Screenshot 2024-03-04 at 08 41 57

**Review by Te-Z — APPROVED** [review]

File: lib/widgets/matrix.dart [hunk 4]
```diff
@@ -591,9 +599,18 @@ class MatrixState extends State<Matrix>
     }
   }
 
+  void onFocus(html.Event e) {
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:602]: onWindowFocus

File: lib/widgets/matrix.dart [hunk 5]
```diff
@@ -591,9 +599,18 @@ class MatrixState extends State<Matrix>
...
+    didChangeAppLifecycleState(AppLifecycleState.resumed);
+  }
+
+  void onBlur(html.Event e) {
```
Comment (hoangdat) [inline lib/widgets/matrix.dart:606]: onWindowBlur

**Review by hoangdat — APPROVED** [review]

**Pull request merged**